lib/java_buildpack/framework/sealights_agent.rb

```diff
@@ -0,0 +1,168 @@
+      end
+
+      def get_download_url
+        credentials = @application.services.find_service(FILTER, TOKEN)['credentials']
+        token = credentials[TOKEN]
+        sealights_data = token.split(/\./)[1]
+        decoded_token = Base64.decode64(sealights_data)
+        token_data = JSON.parse(decoded_token)
+        "#{token_data['x-sl-server']}/v2/agents/sealights-java/recommended/download"
+      end
+
+      def get_agent_path
+        @droplet.sandbox
+      end
+
+      def extract_zip(file, target_directory)
+        with_timing "Extracting Sealights Agent to '#{target_directory}'" do
+          FileUtils.mkdir_p(target_directory)
+          shell "unzip -qq #{file} -d #{target_directory} 2>&1"
+          shell "ls -l #{target_directory} 2>&1"
+        end
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#release)
+      def release
+        credentials = @application.services.find_service(FILTER, TOKEN)['credentials']
+        properties = {
+          'sl.token' => credentials[TOKEN]
+        }
+        properties.map { |k, v| @droplet.java_opts.add_system_property(k, v) }
+
+        @logger.info { "Configuration #{@configuration}" }
+
+        # try_add_system_property(TOKEN)
+        try_add_system_property('buildSessionId')
+        try_add_system_property('buildSessionIdFile')
+        try_add_system_property('proxy')
+        try_add_system_property('labId')
+
+        full_path = File.join(get_agent_path, 'sl-test-listener.jar')
+        agent_path = Pathname.new(full_path)
+
+        @droplet.java_opts.add_javaagent(agent_path)
+
+        announce("Agent was configured. Agent path is set to: '#{agent_path}")
+      end
+
+      def try_add_system_property(key)
+        @logger.info { "try_add_system_property #{key}" }
+        @logger.info { " @configuration[key] #{@configuration[key]}" }
+        @logger.info { " @configuration[key] #{!@configuration[key].nil?}" }
+        if !@configuration[key].nil?
+          value = @configuration[key]
+          prop = 'sl.' + key
+          @logger.info { "Adding #{prop}=#{value}" }
+          @droplet.java_opts.add_system_property(prop, value)
+        else
+          @logger.info { "No value for key #{key}" }
+        end
+      end
+
+      protected
+
+      def download_uri(full_url)
+        announce("Downloading Sealights Agent from '#{full_url}'")
+        begin
+          credentials = @application.services.find_service(FILTER, TOKEN)['credentials']
+          token = credentials[TOKEN]
+          uri = URI.parse(full_url)
+          ENV['http_proxy'] = 'http://127.0.0.1:8888' if credentials.key? 'proxy'
+
+          response = Net::HTTP.start(uri.host, uri.port,
+                                     use_ssl: uri.scheme == 'https',
+                                     verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+            request = Net::HTTP::Get.new uri
+            auth = "Bearer #{token}"
+            @logger.info { auth }
```

Comment:
We definitely cannot log this. It looks like it would contain the user's bearer token, so that would leak credentials into the log stream. Please remove this.